presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java

```diff
@@ -913,28 +919,44 @@ protected RowExpression visitLikePredicate(LikePredicate node, Context context)
                 return likeFunctionCall(value, call(getSourceLocation(node), "LIKE_PATTERN", functionResolution.likePatternFunction(), LIKE_PATTERN, pattern, escape));
             }
 
-            RowExpression prefixOrSuffixMatch = generateLikePrefix(value, pattern);
+            RowExpression prefixOrSuffixMatch = generateLikePrefixOrSuffixMatch(value, pattern);
             if (prefixOrSuffixMatch != null) {
                 return prefixOrSuffixMatch;
             }
 
             return likeFunctionCall(value, call(getSourceLocation(node), CAST.name(), functionAndTypeResolver.lookupCast("CAST", VARCHAR, LIKE_PATTERN), LIKE_PATTERN, pattern));
         }
 
-        private RowExpression generateLikePrefix(RowExpression value, RowExpression pattern)
+        private RowExpression generateLikePrefixOrSuffixMatch(RowExpression value, RowExpression pattern)
         {
             if ((value.getType() instanceof VarcharType || value.getType() instanceof CharType) && pattern instanceof ConstantExpression) {
                 Object constObject = ((ConstantExpression) pattern).getValue();
                 if (constObject instanceof Slice) {
                     Slice slice = (Slice) constObject;
                     String patternString = slice.toStringUtf8();
-                    if (patternString.length() > 1 && !patternString.contains("_")) {
-                        if (patternString.indexOf('%') == patternString.length() - 1) {
+                    int matchLength = patternString.length();
+                    if (matchLength > 1 && !patternString.contains("_")) {
```

Comment:
Negate this if and return early. Less indentation is good!

Reply:
> Negate this if and return early. Less indentation is good!

Single return from methods is better!